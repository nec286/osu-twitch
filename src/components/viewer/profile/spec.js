import Inferno from 'inferno';
import { renderIntoDocument, renderToSnapshot, scryRenderedDOMElementsWithTag } from 'inferno-test-utils';
import * as profile from './';

describe('Statistic', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <profile.Statistic value={ 999999 } />
    );
    expect(tree).toMatchSnapshot();
  })
});

describe('Accuracy', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <profile.Accuracy value={ 99.12345 } />
    );
    expect(tree).toMatchSnapshot();
  });

  it('displays a formatted value', () => {
    const tree = renderIntoDocument(
      <profile.Accuracy value={ 99.12345 } />
    );
    const tds = scryRenderedDOMElementsWithTag(tree, 'td');
    expect(tds[1].innerHTML).toBe('99.12%');
  });
});

describe('Score', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <profile.Score value={ 999999 } />
    );
    expect(tree).toMatchSnapshot();
  });

  it('displays a formatted value', () => {
    const tree = renderIntoDocument(
      <profile.Score value={ 999999 } />
    );
    const tds = scryRenderedDOMElementsWithTag(tree, 'td');
    expect(tds[1].innerHTML).toBe('999,999');
  });
});

describe('Badges', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <profile.Badges user={ { level: 9 } } />
    );
    expect(tree).toMatchSnapshot();
  });
});


describe('Level', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <profile.Level value={ 99.12345 } />
    );
    expect(tree).toMatchSnapshot();
  });

  it('displays a formatted value', () => {
    const tree = renderIntoDocument(
      <profile.Level value={ 99.12345 } />
    );
    const divs = scryRenderedDOMElementsWithTag(tree, 'div');
    expect(divs[2].innerHTML).toBe('99');
  });
});
