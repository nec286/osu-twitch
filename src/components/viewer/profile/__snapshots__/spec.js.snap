// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accuracy renders OK 1`] = `
<tr
  value={99.999}
/>
`;

exports[`CountryPPRank renders OK 1`] = `
<tr
  value={1}
/>
`;

exports[`GlobalPPRank renders OK 1`] = `
<tr
  value={1}
/>
`;

exports[`LetterRank renders OK 1`] = `
<tr
  value={999}
/>
`;

exports[`Level renders OK 1`] = `
<tr
  value={99.99}
/>
`;

exports[`LevelProgress renders OK 1`] = `
<tr
  level={99.99}
/>
`;

exports[`PlayCount renders OK 1`] = `
<tr
  value={1}
/>
`;

exports[`RankedScore renders OK 1`] = `
<tr
  value={999999}
/>
`;

exports[`TotalScore renders OK 1`] = `
<tr
  value={999999}
/>
`;
